templates/vue/src/components/SettingsModal.vue

```diff
@@ -98,23 +193,90 @@ export default {
         backgroundUrl = "",
         autoLayout = false,
         nodeDraggable = true,
+        defaultPermissions = {
+          public: ["read"],
+          authenticated: ["read"],
+        },
+        showAccess = true,
       } = this.settings
       this.backgroundUrl = backgroundUrl
       this.autoLayout = autoLayout
       this.nodeDraggable = nodeDraggable
+      this.defaultPermissions = defaultPermissions
+      this.showAccess = showAccess
     },
     async updateSettings() {
       const settings = Object.assign(this.settings, {
         backgroundUrl: this.backgroundUrl,
         autoLayout: this.autoLayout,
         nodeDraggable: this.nodeDraggable,
+        defaultPermissions: this.defaultPermissions,
+        showAccess: this.showAccess,
       })
       await this.$store.dispatch("updateSettings", settings)
       // TODO: Improve behavior so refresh is not required (currently auto-layout and setting the background image only happen initially)
       // this.$emit("settings-updated", settings);
       // this.closeModal();
       location.reload()
     },
+    updatePermissions(value, rowName, type) {
+      if (rowName.startsWith("user") || wpData.roles.hasOwnProperty(rowName)) {
+        return this.changeIndividualPermission(value, rowName, type)
+      }
+      const rowIndex = this.getPermissionRowIndex(rowName)
+      const lowerPriorityPermissions = this.permissionsOrder.slice(rowIndex + 1)
+      lowerPriorityPermissions.forEach(newRow => {
+        this.changeIndividualPermission(value, newRow, type)
+      })
+    },
+    getPermissionRowIndex(rowName) {
+      return this.permissionsOrder.findIndex(thisRow => thisRow === rowName)
+    },
+    isPermissionDisabled(rowName, type) {
```

Comment:
Since this function (and maybe some other ones) are now being used in two different components, can you please turn them into helper functions so that it's only defined in one place? Then you can utilize those helper functions in both components. Again, we want to avoid having to update things in multiple spots in the application if we need to make changes in the future - and also that way there's less code volume.

Reply:
I've moved this function to Helpers, but long term it might be best to turn the entire permissions table into it's own component, seeing as we are repeating the same functionality.